lib/api/redfish-1.0/chassis.js

```diff
@@ -10,49 +10,36 @@ var Promise = injector.get('Promise'); // jshint ignore:line
 var Errors = injector.get('Errors');
 var taskProtocol = injector.get('Protocol.Task');
 var controller = injector.get('Http.Services.Swagger').controller;
+var env = injector.get('Services.Environment');
+var nodeApi = injector.get('Http.Services.Api.Nodes');
+
+var EntityIdFilterDefaults = [ '3', '7', '8', '66' ];
+
+function _getEnclosedSystems(node) {
+    return _(node.relations)
+        .filter({relationType: 'encloses'})
+        .pluck('targets')
+        .flattenDeep()
+        .value();
+}
 
 var dataFactory = function(identifier, dataName) {
     switch(dataName)  {
         case 'ohai':
         case 'dmi':
-            return waterline.nodes.needByIdentifier(identifier)
-            .then(function(node) {
-                return waterline.catalogs.findLatestCatalogOfSource(node.id, dataName);
-            })
-            .then(function (catalogs) {
-                if (_.isEmpty(catalogs)) {
-                    throw new Errors.NotFoundError(
-                        'No Catalogs Found for Source (' + dataName + ').'
-                    );
-                }
-                return catalogs;
-            });
+            return nodeApi.getNodeCatalogSourceById(identifier, dataName);
         case 'system':
-            return waterline.nodes.findOne({ 
-                or: [
-                    {id: identifier}, 
-                    {name: {'contains': identifier}}
-                ]
-            }).then(function(node) {
-                if(!node) {
-                    throw new Errors.NotFoundError('Cound not find node with id ' + identifier);
-                }
+            return nodeApi.getNodeById(identifier)
+            .then(function(node) {
                 return _.filter(node.relations, function(relation) {
                     return relation.relationType === 'encloses';
                 }).map(function(relation) {
                     return relation.targets[0];
```

Comment:
This is returning a reference to any compute node in the enclosure.  Each target is on equal footing, so the first is picked.